nixos/modules/services/web-apps/stirling-pdf.nix

```diff
@@ -40,9 +40,30 @@ in
         Secrets should be added in environmentFiles instead of environment.
       '';
     };
+
+    dataDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/stirling-pdf";
+      description = "Where stirling-pdf stores persistent files";
+    };
   };
 
   config = lib.mkIf cfg.enable {
+
+    users = {
+      users.stirling-pdf = {
+        isSystemUser = true;
+        home = cfg.dataDir;
+        description = "User for running the stirling-pdf service";
+        group = "stirling-pdf";
+      };
+      groups.stirling-pdf = { };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' - stirling-pdf stirling-pdf - -"
```

Comment:
Since the `dataDir` can be set by an option now, could you add an option to allow setting the user and group as well? (So that the owner of `dataDir` can be changed)

Just like other modules, when `user` and `group` are set different from `stirling-pdf`, this module doesn't need to create the extra user or group.

Reply:
Done